src/main/java/seedu/address/logic/commands/TagInfoCommand.java

```diff
@@ -77,10 +89,25 @@ public CommandResult execute(Model model) throws CommandException {
             resultMessage += "following tags cannot be found:\n" + errorTags;
         }
 
-        ReportWindow.setReportMessage(resultMessage);
-        ReportWindow window = new ReportWindow();
-        window.show();
+        if (!this.isTesting) {
+            ReportWindow.setReportMessage(resultMessage);
+            ReportWindow window = new ReportWindow();
+            window.show();
+        }
+
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
 
-        return new CommandResult("Tag Information is generated in a report window.");
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof TagInfoCommand) {
+            TagInfoCommand compareCommand = (TagInfoCommand) o;
+            for (Tag tag : compareCommand.getListOfTags()) {
+                if (!this.getListOfTags().contains(tag)) {
```

Comment:
Is it possible to remove `this`? If I remember correctly, `this` is not encouraged to use in this module, unlike cs2030s